nixos/modules/system/boot/plymouth.nix

```diff
@@ -65,6 +73,14 @@ in
         '';
       };
 
+      deviceOrientation = mkOption {
```

Comment:
I would suggest that you use instead the `extraConfig` pattern. See `nixos/modules/config/pulseaudio.nix` for a example.
This is for 3 reason: 

 1.  Plain / unpatched plymouth version does not (yet) have the `DeviceOrientation` config field which means your module won't be compatible with unpatched version of the package.
 2.  The extra config pattern has a pretty good chance of being accepted as a nixpkgs pull request.
 3.  This would allow for other kind of config changes we might eventually need.

Reply:
Note that the "typed" version as is would be a great addition too, but only later / if the plymouth package patch has been upstreamed and once we're certain that all nixpkgs users of the plymouth package use at least the patched version.

Reply:
> 3. This would allow for other kind of config changes we might eventually need.
Point taken, that makes a lot of sense.